integration_tests/integration_test.py

```diff
@@ -0,0 +1,131 @@
+"""
+This script is used to test the integration of the mitx-graders library with the Open edX platform.
+Running a code that uses the functions provided by the library using the safe_exec function, and in the MIT course context,
+to be able to use the python_lib.zip that contains the library.
+"""
+
+import os
+import django
+import logging
+from xmodule.capa.safe_exec import safe_exec
+from xmodule.util.sandboxing import SandboxService
+from xmodule.contentstore.django import contentstore
+from opaque_keys.edx.keys import CourseKey
+
+# Set up Django environment
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "cms.envs.test")
+django.setup()
+
+log = logging.getLogger(__name__)
+
+# Define the code to be executed
+all_code = """
+from mitxgraders import *
+
+# Grading Classes
+
+## Single-input graders
+
+StringGrader(answers="cat")
+FormulaGrader(answers='0')
+NumericalGrader()
+MatrixGrader(
+    answers='x*A*B*u + z*C^3*v/(u*C*v)',
+    variables=['A', 'B', 'C', 'u', 'v', 'z', 'x'],
+    sample_from={
+        'A': RealMatrices(shape=[2, 3]),
+        'B': RealMatrices(shape=[3, 2]),
+        'C': RealMatrices(shape=[2, 2]),
+        'u': RealVectors(shape=[2]),
+        'v': RealVectors(shape=[2]),
+        'z': ComplexRectangle()
+    },
+    identity_dim=2
+)
+SingleListGrader(
+    answers=['cat', 'dog', 'unicorn'],
+    subgrader=StringGrader()
+)
+
+## Multi-input graders
+
+ListGrader(
+    answers=['cat', 'dog', 'unicorn'],
+    subgraders=StringGrader()
+)
+
+## Specialized graders
+
+IntegralGrader(
+    answers={
+        'lower': 'a',
+        'upper': 'b',
+        'integrand': 'x^2',
+        'integration_variable': 'x'
+    },
+    input_positions={
+        'integrand': 1,
+        'lower': 2,
+        'upper': 3
+    },
+    variables=['a', 'b']
+)
+
+IntervalGrader(answers=['(','1','2',']'])
+
+SumGrader(
+    answers={
+        'lower': 'a',
+        'upper': 'b',
+        'summand': 'x^2',
+        'summation_variable': 'x'
+    },
+    input_positions={
+        'summand': 1,
+        'lower': 2,
+        'upper': 3
+    },
+    variables=['a', 'b']
+)
+
+"""
+
+
+def execute_code(all_code, course_key_str):
```

Comment:
We could drop this argument if all_code is a constant:
 ```suggestion
def execute_code(course_key_str):
```